--- daos-2.3.105-tb-old/site_scons/env_modules.py	2025-01-14 10:21:49.183938830 +0800
+++ daos-2.3.105-tb/site_scons/env_modules.py	2025-01-14 10:21:26.311938830 +0800
@@ -34,7 +34,7 @@ class _env_module():  # pylint: disable=
     env_module_init = None
     _mpi_map = {"mpich": ['mpi/mpich-x86_64', 'gnu-mpich'],
                 "openmpi": ['mpi/mlnx_openmpi-x86_64', 'mpi/openmpi3-x86_64',
-                            'gnu-openmpi', 'mpi/openmpi-x86_64']}
+                            'gnu-openmpi', 'mpi/openmpi-x86_64', 'mpi/openmpi-aarch64']}
 
     def __init__(self):
         """Load Modules for initializing environment variables"""
