diff --git a/cmake/external.cmake b/cmake/external.cmake
index 2b3e51e076..647c8e26ee 100644
--- a/cmake/external.cmake
+++ b/cmake/external.cmake
@@ -940,7 +940,7 @@ if(NOT ${TD_WINDOWS})       # {
         CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:STRING=${_ins}
         CONFIGURE_COMMAND
             # COMMAND ./Configure --prefix=$ENV{HOME}/.cos-local.2 no-shared
-            COMMAND ./configure --prefix=${_ins} --with-ssl=${ext_ssl_install}
+            COMMAND CFLAGS+=-fPIC CXXFLAGS+="-fPIC" ./configure --prefix=${_ins} --with-ssl=${ext_ssl_install}
                     --enable-websockets --enable-shared=no --disable-ldap
                     --disable-ldaps --without-brotli --without-zstd
                     --without-libidn2 --without-nghttp2 --without-libpsl
