diff --git a/dpdkbuild/Makefile b/dpdkbuild/Makefile
index 64da6cc32..62407e58b 100644
--- a/dpdkbuild/Makefile
+++ b/dpdkbuild/Makefile
@@ -13,14 +13,14 @@ DPDK_OPTS = -Denable_docs=false
 DPDK_OPTS += -Dtests=false

 ifeq ($(shell uname -m),aarch64)
-ifeq ($(shell uname -p),unknown)
+#ifeq ($(shell uname -p),unknown)
 # Unknown ARM platform. ARM doesn't do platform=native which is DPDK's default. So change it.
+#DPDK_OPTS += -Dplatform=generic
+#else
+#ifeq ($(shell $(SPDK_ROOT_DIR)/dpdk/config/arm/armv8_machine.py | cut -d' ' -f4),0x0)
 DPDK_OPTS += -Dplatform=generic
-else
-ifeq ($(shell $(SPDK_ROOT_DIR)/dpdk/config/arm/armv8_machine.py | cut -d' ' -f4),0x0)
-DPDK_OPTS += -Dplatform=generic
-endif
-endif
+#endif
+#endif
 endif

 DPDK_KMODS = false
