diff --git a/BUILD.bazel b/BUILD.bazel
index 834b703f3..1394f1fc3 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -106,6 +106,15 @@ nfpm_pkg(
     visibility = ["//visibility:public"],
 )
 
+nfpm_pkg(
+    name = "kong_oe2203",
+    config = "//build:package/nfpm.yaml",
+    env = nfpm_env,
+    packager = "rpm",
+    pkg_name = "kong.3.4.0.oe2203SP1",
+    visibility = ["//visibility:public"],
+)
+
 ###### flags
 
 # --//:debug=true
diff --git a/Makefile b/Makefile
index 58ece1911..ba84fb505 100644
--- a/Makefile
+++ b/Makefile
@@ -113,10 +113,11 @@ package/apk: check-bazel build-release
 	$(BAZEL) build --config release :kong_apk
 
 package/rpm: check-bazel build-release
-	$(BAZEL) build --config release :kong_el8 --action_env=RPM_SIGNING_KEY_FILE --action_env=NFPM_RPM_PASSPHRASE
-	$(BAZEL) build --config release :kong_el7 --action_env=RPM_SIGNING_KEY_FILE --action_env=NFPM_RPM_PASSPHRASE
-	$(BAZEL) build --config release :kong_aws2	--action_env=RPM_SIGNING_KEY_FILE --action_env=NFPM_RPM_PASSPHRASE
-	$(BAZEL) build --config release :kong_aws2022 --action_env=RPM_SIGNING_KEY_FILE --action_env=NFPM_RPM_PASSPHRASE
+	$(BAZEL) build --config release :kong_oe2203 --action_env=RPM_SIGNING_KEY_FILE --action_env=NFPM_RPM_PASSPHRASE
+#	$(BAZEL) build --config release :kong_el8 --action_env=RPM_SIGNING_KEY_FILE --action_env=NFPM_RPM_PASSPHRASE
+#	$(BAZEL) build --config release :kong_el7 --action_env=RPM_SIGNING_KEY_FILE --action_env=NFPM_RPM_PASSPHRASE
+#	$(BAZEL) build --config release :kong_aws2	--action_env=RPM_SIGNING_KEY_FILE --action_env=NFPM_RPM_PASSPHRASE
+#	$(BAZEL) build --config release :kong_aws2022 --action_env=RPM_SIGNING_KEY_FILE --action_env=NFPM_RPM_PASSPHRASE
 
 functional-tests: dev test
 
diff --git a/diff.patch b/diff.patch
new file mode 100644
index 000000000..e69de29bb
