diff --git a/cmake/external.cmake b/cmake/external.cmake
index 2b3e51e076..647c8e26ee 100644
--- a/cmake/external.cmake
+++ b/cmake/external.cmake
@@ -894,7 +894,7 @@ if(NOT ${TD_WINDOWS})       # {
         CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:STRING=${_ins}
         CONFIGURE_COMMAND
             # COMMAND ./Configure --prefix=$ENV{HOME}/.cos-local.2 no-shared
-            COMMAND ./configure --prefix=${_ins} --with-ssl=${ext_ssl_install}
+            COMMAND CFLAGS+=-fPIC CXXFLAGS+="-fPIC" ./configure --prefix=${_ins} --with-ssl=${ext_ssl_install}
                     --enable-websockets --enable-shared=no --disable-ldap
                     --disable-ldaps --without-brotli --without-zstd
                     --without-libidn2 --without-nghttp2 --without-libpsl
@@ -1099,7 +1099,7 @@ if (${BUILD_CONTRIB} OR NOT ${TD_LINUX})         # {
     endif()
     INIT_EXT(ext_rocksdb
         INC_DIR          include
-        LIB              lib/${ext_rocksdb_static}
+        LIB              lib64/${ext_rocksdb_static}
     )
     # URL https://github.com/facebook/rocksdb/archive/refs/tags/v8.1.1.tar.gz
     # URL_HASH MD5=3b4c97ee45df9c8a5517308d31ab008b